intellij-plugin-verifier/verifier-core/src/main/java/com/jetbrains/pluginverifier/verifiers/method/KotlinMethods.kt

```diff
@@ -0,0 +1,131 @@
+package com.jetbrains.pluginverifier.verifiers.method
+
+import com.jetbrains.pluginverifier.results.location.MethodLocation
+import com.jetbrains.pluginverifier.verifiers.resolution.Method
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.tree.AbstractInsnNode
+import org.objectweb.asm.tree.MethodInsnNode
+
+object KotlinMethods {
+  private const val capacity = 10
+  private val cache = object : LinkedHashMap<MethodLocation, Boolean>(10) {
+    override fun removeEldestEntry(eldest: MutableMap.MutableEntry<MethodLocation, Boolean>?): Boolean {
+      return size > capacity
+    }
+  }
+
+  /**
+   * Identify a Kotlin default method.
+   *
+   * A kotlin default method is a method that has a default implementation, but kotlin compiles
+   * in the inheritor an override that calls the default implementation (inner) class known as
+   * `DefaultImpls`.
+   *
+   * Ignoring labels, line numbers, and kotlin intrinsics (to check nullness), there's only a
+   * handful of interesting bytecode that invoke the static method in `{TheInterfaceType}$DefaultImpls`
+   * loading first `this`, then loading parameters if any.
+   *
+   * E.g. with
+   *
+   * ```
+   * // access flags 0x1
+   * public internalArgsReturningInternal(Linternal/defaultMethod/AnInternalType;Ljava/lang/String;)Linternal/defaultMethod/AnInternalType;
+   *    L0
+   *     ALOAD 1
+   *     LDC "anInternalType"
+   *     INVOKESTATIC kotlin/jvm/internal/Intrinsics.checkNotNullParameter (Ljava/lang/Object;Ljava/lang/String;)V
+   *     ALOAD 2
+   *     LDC "s"
+   *     INVOKESTATIC kotlin/jvm/internal/Intrinsics.checkNotNullParameter (Ljava/lang/Object;Ljava/lang/String;)V
+   *    L1
+   *     LINENUMBER 5 L1
+   *     ALOAD 0
+   *     ALOAD 1
+   *     ALOAD 2
+   *     INVOKESTATIC internal/defaultMethod/InterfaceWithDefaultMethodUsingInternalAPI$DefaultImpls.internalArgsReturningInternal (Linternal/defaultMethod/InterfaceWithDefaultMethodUsingInternalAPI;Linternal/defaultMethod/AnInternalType;Ljava/lang/String;)Linternal/defaultMethod/AnInternalType;
+   *     ARETURN
+   *    L2
+   *     LOCALVARIABLE this Linternal/defaultMethod/InterfaceWithDefaultMethodUsingInternalAPI; L0 L2 0
+   *     LOCALVARIABLE anInternalType Linternal/defaultMethod/AnInternalType; L0 L2 1
+   *     LOCALVARIABLE s Ljava/lang/String; L0 L2 2
+   *     MAXSTACK = 3
+   *     MAXLOCALS = 3
+   * ```
+   *
+   * @see [`JvmDefault` annotation](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/jvm/src/kotlin/jvm/JvmDefault.kt)
+   */
+  fun Method.isKotlinDefaultMethod(): Boolean {
+    val method: Method = this
+    return cache.computeIfAbsent(method.location) {
+      return@computeIfAbsent isKotlinMethodInvokingDefaultImpls(method)
```

Comment:
Sorry, but don't understand why we need `return@computeIfAbsent`

Reply:
Otherwise IJ reports a compilation error

> [RETURN_NOT_ALLOWED] 'return' is not allowed here

Reply:
Hm, couldn't get this error, that's why asking :)

Reply:
I tried compiling from gralde as well but the compiler still wants the label on the `return`.

```
> Task :intellij-plugin-verifier:verifier-core:compileKotlin FAILED
e: file:///Users/brice.dutheil/opensource/intellij-plugin-verifier/intellij-plugin-verifier/verifier-core/src/main/java/com/jetbrains/pluginverifier/verifiers/method/KotlinMethods.kt:60:7 'return' is not allowed here
```